TAMU CSCE Capstone Pull Request 11/3/2024

antonhugo1 opened the pull request:
Pull Request with current progress on CADVentory as of 11/3/2024.

Features:

- Report Generation with customization
- Model Previews in Library Window
- Storing directory information, tops object, and other information in  SQLite database
- some testing

next goals/future improvement:
- Finetuning Report Generation Formatting (landscape, text formatting, etc)
- Integration Testing
- and more

torfoes committed:
Added scripts directory for experiements and utilities.

KotdaPK committed:
base

KotdaPK committed:
stable build for tags display

KotdaPK committed:
CRUD on model properties implemented

KotdaPK committed:
duplicate property inserts return false

KotdaPK committed:
separated the loading of the database from the construtor to a separate function

Commit:
trying to use process g files

KotdaPK committed:
merging step 1

Commit:
png images now go to previews folder

HuigangQu committed:
preview added, remember to modify the ProcessGFiles mged and rt path

KotdaPK committed:
prevents loading of models when they are already loaded

KotdaPK committed:
Revert "prevents loading of models when they are already loaded"

This reverts commit dcec051ea92924e0f624093f13f8fc63e7688ef3.

KotdaPK committed:
display files and doesn't reload models

KotdaPK committed:
working merge

KotdaPK committed:
added insertion before interting properties

HuigangQu committed:
merged from explore_UI

HuigangQu committed:
selecting files for report

antonhugo1 committed:
report generation functionality (need to test more)

torfoes committed:
#define MGED_EXECUTABLE_PATH "@MGED_EXECUTABLE@"
#define RT_EXECUTABLE_PATH "@RT_EXECUTABLE@"

torfoes committed:
include gist

torfoes committed:
init background processing and simplified model interface

torfoes committed:
background indexing with mvc init

torfoes committed:
signaling to stop the background processing

torfoes committed:
model card improvements and init objects selected

torfoes committed:
basic tops object storage test

torfoes committed:
Update model to work on indexing

torfoes committed:
removed db access form indexing worker

torfoes committed:
MVC current

HuigangQu committed:
reload

HuigangQu committed:
fixing

torfoes committed:
init geometry browser

torfoes committed:
Merge pull request #17 from torfoes/model-view-init

model-view-init with geometry selector

torfoes committed:
Delete CMakeLists.txt.user.92a048c

torfoes committed:
Delete CMakeLists.txt.user

HuigangQu committed:
adding model control functions

HuigangQu committed:
merging

Commit:
initial test coverage

Commit:
Added more coverage for Model.cpp to 61.1%. Still needs work

torfoes committed:
onGenerateReportButtonClicked logging selected models and objexcts

torfoes committed:
getSelectedModels, getSelectedObjectsForModel

torfoes committed:
Cleanuped

torfoes committed:
fixed case for librarywindow

KotdaPK committed:
library tests

Commit:
attempting testing for MainWindow

Commit:
fixed issue with parsing in geometry browser

KotdaPK committed:
added tags and simplified executions option

KotdaPK committed:
added print model

torfoes committed:
Merge pull request #18 from torfoes/test-coverage-1

Test coverage 1

torfoes committed:
Single object selection in geometry browser

torfoes committed:
Merge branch 'main' into fix_object_selection

torfoes committed:
Merge pull request #19 from torfoes/fix_object_selection

Fix object selection

torfoes committed:
simplified main cmake

torfoes committed:
init ReportGenerationWindow

torfoes committed:
std::tuple<bool, std::string> ProcessGFiles::generateGistReport(const std::string& inputFilePath, const std::string& outputFilePath) {

torfoes committed:
std::tuple<bool, std::string> generateGistReport(const std::string& inputFilePath, const std::string& outputFilePath);

torfoes committed:
init report generation

torfoes committed:
is_selected, and is_processed

torfoes committed:
modelData.is_processed = true;

torfoes committed:
set(CMAKE_AUTOUIC ON)
set(CMAKE_AUTOMOC ON)

HuigangQu committed:
added settings window

torfoes committed:
../ModelFilterProxyModel.cpp

KotdaPK committed:
working model view test connection

Commit:
added ability to test MainWindow.cpp. We are at 71.5% coverage.

antonhugo1 committed:
attempt to add background processing, added
specific tops component to generate gist on

Commit:
adding basic testing for library window. There is a known issue with genearting the ui_mainwindow.h when building

Commit:
added testing for the GeometryBrowser Dialog, got to 97.9% line coverage and 100% function coverage

antonhugo1 committed:
Report Generation Window, properly non blocking
,progress bar (will need to polish a bit more)

antonhugo1 committed:
Merge pull request #20 from torfoes/pdf_report_gen_branch

Pdf report gen branch

KotdaPK committed:
model view displays and tags are mutable

KotdaPK committed:
working property editing, added library name to model, and reformatted ProcessGFiles with debugging

Commit:
library.cpp test coverage at 100%, Model.cpp at 82% and ProcessGFiles.cpp at 60%

antonhugo1 committed:
landscape cover page, gist reports. working On
table of contents. moved model.working files into
hidden directory after generating report

HuigangQu committed:
merging with settings window, and fixed ui and menubar

HuigangQu committed:
fixed thread quit caused crashing

KotdaPK committed:
added geometry browser and primary file choosing without logic

KotdaPK committed:
removed the primary change functionality

Commit:
Added testing for IndexingWorker, ProcessGFiles. Our test coverage for these files is 52% and 82%

HuigangQu committed:
added settings window

KotdaPK committed:
removed comments to wrap geometrybrowser

torfoes committed:
removed bool preview

torfoes committed:
updated to use QSettings

torfoes committed:
Merge branch 'main' into test-coverage-2

torfoes committed:
Merge pull request #21 from torfoes/test-coverage-2

Test coverage 2

salteradam committed:
Update README.md

Readme now has directions for testing

torfoes committed:
fix TEST_CASE("ProcessGFiles - Generate Gist Report for annual_gift_man.g", "[ProcessGFiles]") {

torfoes committed:
Merge remote-tracking branch 'origin/main'

torfoes committed:
Added is_included field

antonhugo1 committed:
adjusted Table of Contents for landscape

HuigangQu committed:
added preview timeout timer in settings window

antonhugo1 committed:
added qrc/brlcad logo default, classified button
pre-report generation checking for working files

antonhugo1 committed:
stop worker threads in destructor, classified
functionality

brlcad commented:
Hi, just tried compiling on Mac and am seeing a variety of coding issues to resolve.  They mostly seem like they're probably minor issues  -- missing headers, incorrect init ordering, unused variables and such -- but need to be reviewed to make sure some other issue isn't masked.

```
[  6%] Building CXX object CMakeFiles/cadventory.dir/src/CADventory.cpp.o
[  7%] Building CXX object CMakeFiles/cadventory.dir/src/FilesystemIndexer.cpp.o
[  7%] Building CXX object CMakeFiles/cadventory.dir/src/MainWindow.cpp.o
[  8%] Building CXX object CMakeFiles/cadventory.dir/src/SplashDialog.cpp.o
[  8%] Building CXX object CMakeFiles/cadventory.dir/src/Model.cpp.o
[  9%] Building CXX object CMakeFiles/cadventory.dir/src/Library.cpp.o
[  9%] Building CXX object CMakeFiles/cadventory.dir/src/LibraryWindow.cpp.o
/Volumes/X10/cadventory/src/LibraryWindow.cpp:33:5: warning: initializer order does not match the declaration order [-Wreorder-ctor]
    library(nullptr),
    ^~~~~~~~~~~~~~~~
    mainWindow(nullptr)
/Volumes/X10/cadventory/src/LibraryWindow.cpp:33:5: note: field 'library' will be initialized after field 'mainWindow'
    library(nullptr),
    ^~~~~~~~~~~~~~~~
/Volumes/X10/cadventory/src/LibraryWindow.cpp:39:5: note: field 'indexingWorker' will be initialized after field 'modelCardDelegate'
    indexingWorker(nullptr),
    ^~~~~~~~~~~~~~~~~~~~~~~
/Volumes/X10/cadventory/src/LibraryWindow.cpp:256:42: warning: unused parameter 'modelId' [-Wunused-parameter]
void LibraryWindow::onModelProcessed(int modelId) {
                                         ^
2 warnings generated.
[ 10%] Building CXX object CMakeFiles/cadventory.dir/src/ProcessGFiles.cpp.o
[ 10%] Building CXX object CMakeFiles/cadventory.dir/src/IndexingWorker.cpp.o
[ 11%] Building CXX object CMakeFiles/cadventory.dir/src/ModelCardDelegate.cpp.o
/Volumes/X10/cadventory/src/ModelCardDelegate.cpp:78:16: warning: unused variable 'itemPos' [-Wunused-variable]
        QPoint itemPos = pos - itemRect.topLeft();
               ^
1 warning generated.
[ 11%] Building CXX object CMakeFiles/cadventory.dir/src/ModelFilterProxyModel.cpp.o
/Volumes/X10/cadventory/src/ModelFilterProxyModel.cpp:18:14: warning: 'type' is deprecated: Use typeId() or metaType(). [-Wdeprecated-declarations]
    if (data.type() == QVariant::Bool) {
             ^
/opt/homebrew/include/QtCore/qvariant.h:431:5: note: 'type' has been explicitly marked deprecated here
    QT_DEPRECATED_VERSION_X_6_0("Use typeId() or metaType().")
    ^
/opt/homebrew/include/QtCore/qtdeprecationmarkers.h:127:44: note: expanded from macro 'QT_DEPRECATED_VERSION_X_6_0'
# define QT_DEPRECATED_VERSION_X_6_0(text) QT_DEPRECATED_X(text)
                                           ^
/opt/homebrew/include/QtCore/qtdeprecationmarkers.h:27:33: note: expanded from macro 'QT_DEPRECATED_X'
#  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
                                ^
/opt/homebrew/include/QtCore/qcompilerdetection.h:999:36: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#  define Q_DECL_DEPRECATED_X(x) [[deprecated(x)]]
                                   ^
/Volumes/X10/cadventory/src/ModelFilterProxyModel.cpp:18:34: warning: 'Bool' is deprecated: Use QMetaType::Type instead. [-Wdeprecated-declarations]
    if (data.type() == QVariant::Bool) {
                                 ^
/opt/homebrew/include/QtCore/qvariant.h:142:10: note: 'Type' has been explicitly marked deprecated here
    enum QT_DEPRECATED_VERSION_X_6_0("Use QMetaType::Type instead.") Type
         ^
/opt/homebrew/include/QtCore/qtdeprecationmarkers.h:127:44: note: expanded from macro 'QT_DEPRECATED_VERSION_X_6_0'
# define QT_DEPRECATED_VERSION_X_6_0(text) QT_DEPRECATED_X(text)
                                           ^
/opt/homebrew/include/QtCore/qtdeprecationmarkers.h:27:33: note: expanded from macro 'QT_DEPRECATED_X'
#  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
                                ^
/opt/homebrew/include/QtCore/qcompilerdetection.h:999:36: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#  define Q_DECL_DEPRECATED_X(x) [[deprecated(x)]]
                                   ^
2 warnings generated.
[ 12%] Building CXX object CMakeFiles/cadventory.dir/src/GeometryBrowserDialog.cpp.o
[ 12%] Building CXX object CMakeFiles/cadventory.dir/src/ReportGenerationWindow.cpp.o
/Volumes/X10/cadventory/src/ReportGenerationWindow.cpp:25:7: warning: initializer order does not match the declaration order [-Wreorder-ctor]
      model(model),
      ^~~~~~~~~~~~
      num_file(0)
/Volumes/X10/cadventory/src/ReportGenerationWindow.cpp:25:7: note: field 'model' will be initialized after field 'library'
      model(model),
      ^~~~~~~~~~~~
/Volumes/X10/cadventory/src/ReportGenerationWindow.cpp:26:7: note: field 'library' will be initialized after field 'num_file'
      library(library),
      ^~~~~~~~~~~~~~~~
/Volumes/X10/cadventory/src/ReportGenerationWindow.cpp:60:22: error: implicit instantiation of undefined template 'std::basic_ostringstream<char>'
  std::ostringstream oss;
                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk/usr/include/c++/v1/iosfwd:131:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_ostringstream;
                               ^
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/cadventory.dir/src/ReportGenerationWindow.cpp.o] Error 1
make[1]: *** [CMakeFiles/cadventory.dir/all] Error 2  
```

brlcad commented on scripts/README.md:
Note, if you have a bash/linux/cli environment, you can achieve a similar result with these 1-liners:

    # non-recursive
    for i in *.asc ; do asc2g $i $i.g ; done
    
    # or recursively
    find . -name \*.asc -exec asc2g {} {}.g \;

and of course a --single conversion just becomes

> asc2g aet.asc aet.g

brlcad commented on src/LibraryWindow.cpp:
```diff
@@ -64,18 +69,23 @@ LibraryWindow::loadFromLibrary(Library* _library)
 }
 
 
-void
```
If you're going to update style on methods, you will need to be prepared to update all of them for consistency.  That means reviewing the changes made and ensuring all source files are using the same style formatting.

brlcad commented on src/Model.cpp:
```diff
@@ -1,67 +1,88 @@
+    std::string sqlModels = R"(
         CREATE TABLE IF NOT EXISTS models (
             id INTEGER PRIMARY KEY AUTOINCREMENT,
-            short_name TEXT NOT NULL UNIQUE,
```
What's the motivation for this?  The short name ideally needs to be unique to avoid a slew of UX confusion possibilities.

Commit:
No longer measuring coverage of MainWindow and LibraryWindow for now due to filesystem errors during testing. Added more tests for IndexingWorker but there is a slight issue with how fast it runs

torfoes committed:
is processed role

torfoes committed:
../FileSystemModelWithCheckboxes.cpp
    ../FileSystemModelWithCheckboxes.h
    ../FileSystemFilterProxyModel.cpp
    ../FileSystemFilterProxyModel.h

torfoes committed:
init filesystem view

torfoes committed:
visual filesystem

KotdaPK committed:
string error

antonhugo1 committed:
ui and functionality for subtitle, classification

KotdaPK committed:
preview image fix, properties editable indication, and ui change

antonhugo1 committed:
Adjusted table of content pages, error msg bugs

KotdaPK committed:
main and modelCRUD merge

torfoes committed:
filsystemModel with checkbox init database interfacing

torfoes committed:
rm void LibraryWindow::includeAllModels() {

torfoes committed:
id INTEGER PRIMARY KEY AUTOINCREMENT,
short_name TEXT NOT NULL UNIQUE,            file_path TEXT UNIQUE, and     std::lock_guard<std::recursive_mutex> lock(db_mutex);

torfoes committed:
model->insertModel(modelData);
instead of model->insertModel(modelId, modelData);

antonhugo1 committed:
Paging! and with errors, also fixed margins

torfoes committed:
ModelData existingModel = model->getModelByFilePath(file_path.string());

torfoes committed:
QString currentObject = QString::fromStdString(modelData.short_name);

torfoes committed:
logic for checkboxes behaving as expected

HuigangQu committed:
fixed mainwindow size

antonhugo1 committed:
temporarily assigning timeout to be 99999

antonhugo1 committed:
Merge pull request #23 from torfoes/pdf_report_gen_branch

Report Generation Branch Sprint 4

torfoes committed:
removed comment

torfoes committed:
Merge branch 'main' into file-system-view

torfoes committed:
Merge pull request #24 from torfoes/file-system-view

File system view

torfoes committed:
commented out outdated test

KotdaPK committed:
fixed merge issues

KotdaPK committed:
merged modelCRUD

torfoes committed:
init linked processing

antonhugo1 committed:
gear, hierarchy, and settings icons into qrc

antonhugo1 committed:
adjust generate report to include command in error

torfoes committed:
init FindBRLCAD.cmake

torfoes committed:
updated CMakeList.txt with BRLCAD_ROOT

torfoes committed:
simplified version with working extract title

torfoes committed:
commented out gist functionality for now so i can focus on updating processgfiles

torfoes committed:
init object extraction with libged

torfoes committed:
object extraction and improved error handling

torfoes committed:
removed uneeded previews folder

torfoes committed:
rm unused slots/signals

torfoes committed:
init getIncludedNotProcessedModels

antonhugo1 committed:
stop report generation functionality

antonhugo1 committed:
Merge pull request #26 from torfoes/report_gen_fix

added logos to qrc, stop button in report generation, command into error pop up

torfoes committed:
init reindex request

torfoes committed:
init generateGistReport with QProcess

KotdaPK committed:
modelview ui changes

KotdaPK committed:
Save local changes before merging

KotdaPK committed:
Merge branch 'bug_fixes'

KotdaPK committed:
modelview ui improvement

torfoes committed:
updated std::tuple<bool, std::string, std::string> ProcessGFiles::generateGistReport(const std::string& inputFilePath, const std::string& outputFilePath, const std::string& primary_obj, const std::string& label)

torfoes committed:
Merge branch 'main' into using-linked-brlcad-functions

torfoes committed:
Merge pull request #27 from torfoes/using-linked-brlcad-functions

Using linked brlcad functions

torfoes committed:
gistReportTimeout

torfoes committed:
modelData.is_selected = false;

torfoes committed:
is_selected=false

torfoes committed:
is_selected=false

torfoes committed:
new icons!

antonhugo1 committed:
progress bar fix

HuigangQu committed:
removed local home, and added reset tool

HuigangQu committed:
added remove library in file menu

HuigangQu committed:
fixing reload to update GUI

torfoes committed:
Merge branch 'main' into test_coverage_3

torfoes committed:
Merge pull request #25 from torfoes/test_coverage_3

No longer measuring coverage of MainWindow and LibraryWindow for now …

antonhugo1 committed:
properly show error message rather than exit code

Commit:
issues with linking brlcad in test suite

antonhugo1 committed:
comments for report generation functions

Commit:
updated multiple tests like model to better align with new code

Commit:
Added tests for FileSystemFilterProxyModelTest.cpp and FileSystemModelWithCheckboxesTest.cpp

torfoes committed:
update to tests/CMakeLists.txt

HuigangQu committed:
fixed Mainwindowtests

HuigangQu committed:
fixed remove library UI bug

Commit:
Adjusted cmake lists for coverage report

Commit:
added code coverage flags for testing, updated IndexingWorkerTest for more coverage

HuigangQu committed:
Updated Installation section in  README.md

Commit:
ensured all model tests are pasing

HuigangQu committed:
fixed filesystemFilter test and filesystemModel test

torfoes committed:
IndxingWorkerTest and ProcessGfilesTest fix

Commit:
Automation testing now runs via run_tests.py, README now reflects this

torfoes committed:
Merge pull request #28 from torfoes/test-coverage-5

Test coverage 5
